File: Math/9. Palindrome Number/Palindrome Number.py
"""
Solution 2
"""
class Solution(object):
    def isPalindrome(self, x):
        """
        :type x: int
        :rtype: bool
        """
        b = str(x)
        converNum = b[::-1]
        return converNum == b
        

"""
Solution 2
"""
class Solution(object):
    def isPalindrome(self, x):
        """
        :type x: int
        :rtype: bool
        """
        if x<0:
            return False
        a = 0
        b = x
        while b>0:
            a =a*10+b%10
            b = b/10
        return a==x
        
        

File: Math/9. Palindrome Number/Palindrome Number.cpp
//Solution 1
class Solution {
public:
	bool isPalindrome(int x) {
		if (x<0)   //所有的负数都不是回文数
			return false;
		int a = x;
		long b = 0;
		while (a != 0){
			b = b * 10 + a % 10;
			a /= 10;
		}

		if (b == x)
			return true;
		else
			return false;
	}
};

//Solution 2
class Solution {
public:
	bool isPalindrome(int x){
		//特殊情况
		if (x < 0 || (x % 10 == 0 && x != 0)){
			return 0
		}
		long a = 0;
		int b = x;
		while (a < b){
			a = a * 10 + b % 10;
			b /= 10;
		}
		return b = = a || b == a / 10;
	}
};